src/app/components/pages/AdminUserManager.tsx

```diff
@@ -21,48 +41,170 @@ import { TitleAndBreadcrumb } from "../elements/TitleAndBreadcrumb";
+                  <option value="TEACHER">Teacher</option>
+                  <option value="CONTENT_EDITOR">Content editor</option>
+                  <option value="EVENT_LEADER">Event leader</option>
+                  <option value="EVENT_MANAGER">Event manager</option>
+                  <option value="ADMIN">Admin</option>
+                </Input>
+              </FormGroup>
+              <FormGroup>
+                <Label htmlFor="postcode-search">Find users with school within a given distance of postcode:</Label>
+                <Row>
+                  <Col md={7}>
+                    <Input
+                      id="postcode-search"
+                      type="text"
+                      defaultValue={searchQuery.postcode ?? undefined}
+                      placeholder="e.g. CB3 0FD"
+                      onChange={(e: React.ChangeEvent<HTMLInputElement>) => updateQuery({ postcode: e.target.value })}
+                    />
+                  </Col>
+                  <Col md={5} className="mt-2 mt-md-0">
+                    <Input
+                      id="postcode-radius-search"
+                      type="select"
+                      defaultValue={searchQuery.postcodeRadius}
+                      onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+                        updateQuery({ postcodeRadius: e.target.value })
+                      }
+                    >
+                      <option value="FIVE_MILES">5 miles</option>
+                      <option value="TEN_MILES">10 miles</option>
+                      <option value="FIFTEEN_MILES">15 miles</option>
+                      <option value="TWENTY_MILES">20 miles</option>
+                      <option value="TWENTY_FIVE_MILES">25 miles</option>
+                      <option value="FIFTY_MILES">50 miles</option>
+                    </Input>
+                  </Col>
+                </Row>
+              </FormGroup>
+              <FormGroup>
+                <Label htmlFor="school-urn-search">Find a user with school URN:</Label>
+                <Input
+                  id="school-urn-search"
+                  type="text"
+                  defaultValue={searchQuery.schoolURN ?? undefined}
+                  onChange={(e: React.ChangeEvent<HTMLInputElement>) => updateQuery({ schoolURN: e.target.value })}
+                />
+              </FormGroup>
+            </Col>
+          </Row>
+        </CardBody>
+        <CardFooter>
+          <Row>
+            <Col md={{ size: 4, offset: 4 }}>
+              <Input type="submit" value="Search" className="btn btn-block btn-secondary border-0" />
+            </Col>
+          </Row>
+        </CardFooter>
+      </Form>
+    </Card>
+  );
+};
+
+const UserManagerResults = ({ searchRequested, searchQuery }: { searchRequested: boolean; searchQuery: object }) => {
+  const dispatch = useAppDispatch();
+  const searchResults = useAppSelector(selectors.admin.userSearch);
```

Comment:
One thing I tend to do when I deal with a collection, like this array, is that unless it's strictly necessary, I make sure the collection is never `null` or `undefined`. That way, you can make the rest of your code a lot cleaner and simpler, because you know that `searchResults` will always be an array. So, for instance, a few lines below, you wouldn't need to do `isDefined(searchResults)`, because you know for sure that `searchResults` is always an array and never `null` or `undefined`. One way to do this is by changing this line to:
```typescript
const searchResults = useAppSelector(selectors.admin.userSearch) ?? [];
```

There is a different `searchResults` constant in this file that you can also change this way. And you can do this for every collection, really.

Reply:
Updated this, though for the `searchResults` in the `AdminUserManager` I think it works better the way it was already, because adding `?? []` actually causes a rendering error and it wants it to be memoised / contained within useEffect. However that's a bit of a faff because you can't use useSelector inside a callback. So in that instance I'm in favour of leaving it alone!

Reply:
Sounds good, thank you, Hannah.